Python_Backend/dummy_server.py

```diff
@@ -0,0 +1,175 @@
+multi_face_detected_exception = "ERROR: Multiple Faces Detected"
+invalid_model_type = "ERROR: Invalid Model Type"
+
+df_generator_exception = "ERROR: DF Generator Failed"
+state_generator_exception = "ERROR: State Generator Failed"
+answer_generator_exception = "ERROR: Answer Generator Failed"
+
+current_answer = contextvars.ContextVar('current_answer', default=AnswerGenerator.Answer.UNDEFINED)
+
+def get_landmarks(image_data):
+    if (image_data[0] == FACE):
+
+        try: 
+            df = DFGenerator.FacialDFGenerator.generate_df(image_data[1])
+        
+        except DFGenerator.NoFaceDetectedException:
+            return no_face_detected_exception
+        
+        except DFGenerator.MultiFaceDetectedException:
+            return multi_face_detected_exception
+
+        except Exception:
+            return df_generator_exception
+
+
+    elif (image_data[0] == IRIS):
+        
+        try:
+            df = DFGenerator.IrisDFGenerator.generate_df(image_data[1])
+
+        except DFGenerator.NoFaceDetectedException:
+            return no_face_detected_exception
+        
+        except DFGenerator.MultiFaceDetectedException:
+            return multi_face_detected_exception
+
+        except Exception:
+            return df_generator_exception
+
+    else:
+        print("how did this even happen?")
+        return Exception
+
+
+    return df
+
+#function to convert an image that is already in a Pillow image format (jpg) to cv2
+#https://stackoverflow.com/questions/14134892/convert-image-from-pil-to-opencv-format
+def convert_image(im):
+    open_cv_image = numpy.array(im)
+    open_cv_image = open_cv_image[:,:,::-1].copy()
+    return open_cv_image
+
+#async function to recieve an image or images through a websocket
+#https://stackoverflow.com/questions/26070547/decoding-base64-from-post-to-use-in-pil
+#takes in the message from the websocket and then strips out all the unneeded header information so we just have the base64 encoded data
+#this data can then be converted into a PIL image and then dealt with as needed
+async def recv_image(websocket):
+
+    async for message in websocket:
+        #take in message as a json and then get the data from it according to its tags
+        try:
+            as_json = json.loads(message)
+        except json.JSONDecodeError as ex:
+            print("There was an error with the JSON data from the frontend")
+            pass
+        
+        msg_mode = as_json["mode"]
+        if (msg_mode == "face"):
+            mode = FACE
+        elif (msg_mode == "eye"):
+            mode = IRIS
+        #safe default if there is a problem with the mode type
+        else:
+            print("Something is wrong with the recieved mode type, defaulting to face tracking.")
+            mode = FACE
+
+        img_data = as_json["image"]
+
+        temp = img_data.split(",")
+        for i in temp:
+            if(i != "data:image/jpeg;base64"):
```

Comment:
Can you clarify what this section means? Are there multiple images that your processing in one message?